include/tvm/tir/op.h

```diff
@@ -911,7 +911,9 @@ inline PrimExpr MakeConstScalar(DataType t, ValueType value, Span span = Span())
   if (t.is_uint()) {
     // Use IntImm if it is a small integer
     uint64_t uval = static_cast<uint64_t>(value);
-    if (uval <= static_cast<uint64_t>(std::numeric_limits<int64_t>::max())) {
+    if (value < 0) {
```

Comment:
What if `value` is greater than `numeric_limit<int64_t>::max()`? Is it possible to keep the sign information. If it is from Python end, `int` in Python is of unlimited length so catching the sign is easy. The sign information in C++ probably can be catched from the type information, i.e., ValueType.